### PR TITLE
aws_s3/test: keep length of name below 63 char

### PR DESCRIPTION
**Pull request by goneri**
Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/954

To ensure the name of the bucket with ACL is valide we build
the name from the md5 of the `resource_prefix`.

### CODE REVIEW

**Comment by goneri** [comment]
recheck

File: tests/integration/targets/aws_s3/defaults/main.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ---
 # defaults file for s3
 bucket_name: '{{ resource_prefix }}'
```
Comment (abikouo) [inline tests/integration/targets/aws_s3/defaults/main.yml:3]:
```suggestion
bucket_name: '{{ resource_prefix | hash("md5") }}'
```

Comment (abikouo) [inline tests/integration/targets/aws_s3/defaults/main.yml:3]:
```suggestion
bucket_name: '{{ resource_prefix | hash("md5")}}'
```

File: tests/integration/targets/aws_s3/defaults/main.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
-bucket_name_acl: '{{ bucket_name }}-with-acl'
+bucket_name_acl: "{{ resource_prefix | hash('md5') + '-with-acl' }}"
```
Comment (abikouo) [inline tests/integration/targets/aws_s3/defaults/main.yml:4]:
```suggestion
bucket_name_acl: "{{ bucket_name + '-with-acl' }}"
```

**Review by abikouo — APPROVED** [review]
LGTM

**Comment by jillr** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Comment by goneri** [comment]
recheck

**Pull request merged**